src/main/java/org/sonarlint/intellij/config/global/SonarQubeServerMgmtPanel.java

```diff
@@ -333,17 +333,20 @@ private void actionUpdateServerTask(boolean background) {
 
   public static void updateServerBinding(SonarQubeServer server, ConnectedSonarLintEngine engine, boolean background, boolean onlyProjects) {
     Project[] openProjects = ProjectManager.getInstance().getOpenProjects();
-    Set<String> projectKeys = new HashSet<>();
+    List<Project> projects = new ArrayList<>(openProjects.length);
+    List<String> moduleKeys = new ArrayList<>(openProjects.length);
+    Set<String> unique = new HashSet<>();
 
     for (Project p : openProjects) {
       SonarLintProjectSettings projectSettings = SonarLintUtils.get(p, SonarLintProjectSettings.class);
-      if (projectSettings.isBindingEnabled() && server.getName().equals(projectSettings.getServerId())
-        && projectSettings.getProjectKey() != null) {
-        projectKeys.add(projectSettings.getProjectKey());
+      String moduleKey = projectSettings.getProjectKey();
+      if (projectSettings.isBindingEnabled() && server.getName().equals(projectSettings.getServerId()) && moduleKey != null && unique.add(moduleKey)) {
```

Comment:
If 2 projects are bound to the same module -> module key won't be unique, but we still want to analyze both projects. Why not simply use a Set to hold moduleKeys?